lib/cipherstash/protect/active_record_extensions/bloom_filter.rb

```diff
@@ -35,35 +36,59 @@ class BloomFilter
         # @return [Integer]
         attr_reader :k
 
+        # Postgres does not support unsigned ints. The max signed small int
+        # value is 32767.
+        #
+        # To be able to store values up to 2 ^ 16 (65536) we need to offset each
+        # bit position by -32_768 when storing and querying bloom filters.
+        # @return [Integer]
+        attr_reader :postgres_offset
+
         # Creates a new bloom filter with the given key and filter match index settings.
         #
-        # @param key [String] the key to use for hashing terms. Should be provided as a hex-encoded string.
+        # @param id [String] the id that is used with the local key to derive a key for hashing. Should be provided as a uuid.
```

Comment:
I updated the bloom filter to require the id and options only when initializing.

I removed the key as an arg because we derive the key from the CS_PROTECT_KEY combined with the id. The CS_PROTECT_KEY will always be the same. 

Instead we get the cs_protect_key from the rails creds, or env vars within the initialize, then use hkdf with the id to derive the bloom filter key.